core/dive.c

```diff
@@ -3020,30 +3021,30 @@ void taglist_cleanup(struct tag_entry **tag_list)
 	}
 }
 
-int taglist_get_tagstring(struct tag_entry *tag_list, char *buffer, int len)
+char *taglist_get_tagstring(struct tag_entry *tag_list)
 {
-	int i = 0;
-	struct tag_entry *tmp;
-	tmp = tag_list;
-	memset(buffer, 0, len);
+	bool first_tag = true;
+	struct membuffer b = { 0 };
+	struct tag_entry *tmp = tag_list;
 	while (tmp != NULL) {
-		int newlength = strlen(tmp->tag->name);
-		if (i > 0)
-			newlength += 2;
-		if ((i + newlength) < len) {
-			if (i > 0) {
-				strcpy(buffer + i, ", ");
-				strcpy(buffer + i + 2, tmp->tag->name);
+		if (*tmp->tag->name != '\0') {
```

Comment:
You could write this as
```
if (!empty_string(tmp->tag->name))
```
This handles `name` being the null pointer gracefully.